### PR TITLE
docs(python): make "DataFrame" and "Series" case more consistent across docs/comments/errors

### PR DESCRIPTION
**Pull request by alexander-beedie**
We have been a bit inconsistent in our use of "dataframe" vs "DataFrame" and/or "series" vs "Series" in the docs, comments, and errors. This largely cleans it up by standardising on "DataFrame" and "Series", as these are the concrete types being referred to.

### CODE REVIEW

**Comment by stinodego** [comment]
> (Also: removed a note in the docs that says we rely on pyarrow for interchange protocol support, as that hasn't been true since https://github.com/pola-rs/polars/pull/10267 ;)

We do still rely on pyarrow for `from_dataframe`, so that part of the docstring is correct.

File: py-polars/polars/dataframe/frame.py [hunk 1]
```diff
@@ -1285,7 +1285,7 @@ def __dataframe__(
         allow_copy: bool = True,  # noqa: FBT001
     ) -> PolarsDataFrame:
         """
-        Convert to a dataframe object implementing the dataframe interchange protocol.
+        Convert to a DataFrame object implementing the dataframe interchange protocol.
```
Comment (stinodego) [inline py-polars/polars/dataframe/frame.py:1288]: This should remain "dataframe" as it's not referring to the Polars DataFrame class, it's referring to a generic dataframe object.

File: py-polars/polars/dataframe/frame.py [hunk 2]
```diff
@@ -1307,7 +1307,7 @@ def __dataframe__(
 
         Examples
         --------
-        Convert a Polars dataframe to a generic dataframe object and access some
+        Convert a Polars DataFrame to a generic DataFrame object and access some
```
Comment (stinodego) [inline py-polars/polars/dataframe/frame.py:1310]: Same here.

```suggestion
        Convert a Polars DataFrame to a generic dataframe object and access some
```

File: py-polars/polars/interchange/__init__.py [hunk 3]
```diff
@@ -1,5 +1,5 @@
 """
-Module containing the implementation of the Python dataframe interchange protocol.
+Module containing the implementation of the Python DataFrame interchange protocol.
```
Comment (stinodego) [inline py-polars/polars/interchange/__init__.py:2]: Again this must remain lowercase:
https://data-apis.org/dataframe-protocol/latest/index.html

**Review by stinodego — CHANGES_REQUESTED** [review]
For some of these changes I agree, but for most parts of the dataframe interchange protocol this should be reverted. In that context, we are talking about generic dataframe objects and not referring to a specific `DataFrame` class name.

File: py-polars/polars/interchange/__init__.py (hunk 3 above)
Comment (alexander-beedie) [reply py-polars/polars/interchange/__init__.py:2]: Doh; thought I _had_ left all of those alone, for exactly the reason given. A few rogue replacements must have made it through - will fix 🤦

Comment (alexander-beedie) [reply py-polars/polars/interchange/__init__.py]: Updated/reverted these👌

**Comment by alexander-beedie** [comment]
> We do still rely on pyarrow for `from_dataframe`, so that part of the docstring is correct.

Ahh; one direction is native, the other is not; returned that part of the docstring to its original state, with a slight clarification that it is only true for `from_dataframe`.

File: py-polars/polars/interchange/dataframe.py [hunk 4]
```diff
@@ -22,7 +22,7 @@ class PolarsDataFrame(InterchangeDataFrame):
     Parameters
     ----------
     column
-        The Polars DataFrame backing the dataframe object.
+        The Polars DataFrame backing the dataframe interchange object.
```
Comment (stinodego) [inline py-polars/polars/interchange/dataframe.py:25]: Could this please just remain "dataframe object"? Matches line 20.

```suggestion
        The Polars DataFrame backing the dataframe object.
```

File: py-polars/polars/dataframe/frame.py (hunk 1 above)
Comment (stinodego) [inline py-polars/polars/dataframe/frame.py:1288]: Can we keep the casing here - object is technically a `PolarsDataFrame` but that's not important here.

```suggestion
        Convert to a dataframe object implementing the dataframe interchange protocol.
```

**Review by stinodego — CHANGES_REQUESTED** [review]
Sorry for nitpicking these but I actually put some thought into the casing of these docstrings :grimacing:

**Review by stinodego — APPROVED** [review]

**Comment by alexander-beedie** [comment]
> Sorry for nitpicking these

Lol... nitpick to your heart's content; we're making constant forward progress ;)

**Pull request merged**